lib/unleash/constraint.rb

```diff
@@ -45,7 +47,17 @@ def matches_context?(context)
     rescue KeyError
       Unleash.logger.warn "Attemped to resolve a context key during constraint resolution: #{self.context_name} but it wasn't \
       found on the context"
-      false
+      key_error_handler()
+    end
+
+    def key_error_handler()
+      case operator
+      when "NOT_IN".to_sym
+        # when there is no input data present, nothing can't be in anything.
+        true
+      else
+        false
+      end
```

Comment:
```suggestion

      # when the operator is NOT_IN and there is no data, return true. In all other cases the operator doesn't match.
      return true if operator == :NOT_IN
      false
```

Reply:
this could be even simpler by just having:
```suggestion
    (operator == :NOT_IN)
```

But I think that might be too concise and reduce legibility.

Reply:
I like the first suggestion! I think we should rather unpack a bit in the comment why this behaviour is here, because I think this is a surprising edge case in constraints on the Unleash side and its helpful to understand why this code is necessary

Reply:
Agree!